src/Lykke.AlgoStore.Api/Controllers/AlgoController.cs

```diff
@@ -84,6 +84,24 @@ public AlgoController(IAlgosService clientDataService, IAlgoStoreService service
             return Ok(response);
         }
 
+        [HttpGet("getAllUserAlgos")]
+        [SwaggerOperation("GetAllUserAlgos")]
+        [ProducesResponseType(typeof(List<AlgoDataModel>), (int)HttpStatusCode.OK)]
+        [ProducesResponseType(typeof(BaseErrorResponse), (int)HttpStatusCode.InternalServerError)]
+        [ProducesResponseType((int)HttpStatusCode.NotFound)]
+        public async Task<IActionResult> GetAllUserAlgos()
+        {
+            var clientId = User.GetClientId();
+            var result = await _algosService.GetAllUserAlgosAsync(clientId);
+
+            if (result.IsNullOrEmptyCollection())
+                return NotFound();
```

Comment:
I don't think the endpoint should return not found if the collection is empty - it's possible that the user won't have any algos. It's probably better to just return an empty array.